presto-parser/src/main/java/com/facebook/presto/sql/tree/ShowStats.java

```diff
@@ -13,32 +13,50 @@
  */
 package com.facebook.presto.sql.tree;
 
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+
 import java.util.List;
 import java.util.Objects;
 import java.util.Optional;
 
+import static com.facebook.presto.sql.QueryUtil.simpleQuery;
 import static com.google.common.base.MoreObjects.toStringHelper;
-import static java.util.Collections.emptyList;
 
 public class ShowStats
         extends Statement
 {
-    private final QualifiedName table;
+    private final Query query;
 
+    @VisibleForTesting
     public ShowStats(QualifiedName table)
     {
         this(Optional.empty(), table);
     }
 
-    public ShowStats(NodeLocation location, QualifiedName table)
+    @VisibleForTesting
+    public ShowStats(Query query)
+    {
+        this(Optional.empty(), query);
+    }
+
+    public ShowStats(Optional<NodeLocation> location, QualifiedName table)
     {
-        this(Optional.of(location), table);
+        this(location, createFakeQuery(location, table));
     }
 
-    private ShowStats(Optional<NodeLocation> location, QualifiedName table)
+    public ShowStats(Optional<NodeLocation> location, Query query)
     {
         super(location);
-        this.table = table;
+        this.query = query;
+    }
+
+    private static Query createFakeQuery(Optional<NodeLocation> location, QualifiedName name)
```

Comment:
This is hacky. It also makes this tree node not roundtrip very well.

SHOW STATS FROM t   => AST => SHOW STATS FOR (SELECT * FROM t)

A slightly better take would be to use Relation instead of Query. `TABLE t` (Table) and `(SELECT ...)` (TableSubquery) are both `Relation`.